MapboxDirections/MBDirectionsRouter.swift

```diff
@@ -48,83 +49,65 @@ internal enum MBDirectionsRouter: Router {
         case Full = "full"
     }
     
-    case V4(Configuration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
-    case V5(Configuration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
-    
-    var method: HTTPMethod {
-        return .GET
-    }
-    
-    var encoding: HTTPEncoding {
-        return .URL
-    }
-    
-    var configuration: Configuration {
-        switch self {
-        case .V4(let config, _, _, _, _, _, _): return config
-        case .V5(let config, _, _, _, _, _, _, _): return config
-        }
-    }
+    case V4(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
+    case V5(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
```

Comment:
Right now both versions need to be supported until the three default profiles are available for v5. But I think we could also find ourselves in a situation down the road where multiple endpoints need to be supported simultaneously. Originally, I wanted to keep completely separate code paths but expose them as one unified API using class clusters. Unfortunately, Swift currently lacks language support for Objective-C-style class clusters.